api/src/main/java/org/apache/iceberg/util/DateTimeUtil.java

```diff
@@ -120,19 +176,32 @@ public static long isoTimeToMicros(String timeString) {
   }
 
   public static long isoTimestamptzToMicros(String timestampString) {
-    return microsFromTimestamptz(
-        OffsetDateTime.parse(timestampString, DateTimeFormatter.ISO_DATE_TIME));
+    return microsFromTimestamptz(isoTimestamptzToOffsetDateTime(timestampString));
+  }
+
+  public static OffsetDateTime isoTimestamptzToOffsetDateTime(String timestamp) {
+    return OffsetDateTime.parse(timestamp, DateTimeFormatter.ISO_DATE_TIME);
+  }
+
+  public static LocalDateTime isoTimestampToLocalDateTime(String timestamp) {
```

Comment:
Is this needed? If not, just expose `isoTimestamptzToNanos`.

Reply:
Reworked.